caffeine/src/main/java/com/github/benmanes/caffeine/cache/BoundedLocalCache.java

```diff
@@ -3156,7 +3166,8 @@ Object writeReplace() {
       @Override public long getMaximum() {
         return cache.maximum();
       }
-      @Override public void setMaximum(long maximum) {
+      @Override public void setMaximum(long maximum){ setMaximum(maximum, PERCENT_MAIN);}
+      @Override public void setMaximum(long maximum, double percentMain) {
         cache.evictionLock.lock();
         try {
           cache.setMaximum(maximum);
```

Comment:
Should this be setting the `percentMain` too ?  Anyway we can invent a test, with a really small cache, to check?